R/SEIRDAge_interventions.R

```diff
@@ -0,0 +1,452 @@
+    S0 = value$S0
+    E0 = value$E0
+    I0 = value$I0
+    R0 = value$R0
+    D0 = value$D0
+    # check that ICs are valid
+    if (sum(S0, E0, I0, R0, D0) != 1) {
+      stop('Invalid initial conditions. Must sum to 1.')
+    }
+    
+    # create list of parameter values
+    ic <- list(S0, E0, I0, R0, D0)
+    
+    # add names to each value
+    names(ic) = object@initial_condition_names
+    
+    # raise errors if age category dimensions do not match initial state vectors
+    # also raise errors if initial state and parameter values are not doubles
+    for (p in list('S0', 'E0', 'I0', 'R0', 'D0')){
+      if(length(ic[[p]]) != object@n_age_categories){
+        stop(glue('Wrong number of age groups for {p}
+              compartments.'))}
+      if(!is.numeric(ic[[p]])){
+        stop(glue('{p} format must be numeric'))}
+    }
+    if(sum(S0, E0, I0, R0, D0) != 1){
+      stop('All compartments need to sum up to 1.')
+    }
+    
+    # if all above tests are passed, assign the ic namelist to the object
+    object@initial_conditions <- ic
+    
+    return(object)
+  })
+
+# Setter and getter methods for transmission_parameters of an age-structured
+# SEIRD model with interventions interventions
+
+#' @describeIn SEIRDAge_interventions Retrieves transmission_parameters for an
+#' age-structured SEIR model with interventions.
+#'
+#' @param object An object of the class SEIRDAge_interventions.
+#'
+#' @return Transmission parameters of SEIRDAge_interventions model.
+#'
+#' @export
+setMethod('transmission_parameters', 'SEIRDAge_interventions',
+          function(object) object@transmission_parameters)
+
+
+#' @describeIn SEIRDAge_interventions Sets the transmission_parameters for an
+#' age-structured SEIRD model with interventions.
+#' 
+#' Sets transmission_parameters of an
+#' age-structured SEIR model.
+#'
+#' If the transmission parameters provided to are not 1-dimensional an error is
+#' thrown.
+#'
+#' @param value a named list of form list(beta=, kappa=, gamma=, mu=)
+#'
+#' All rates of change between compartments are equal regardless of
+#' age group.
+#'
+#' @return Updated version of the age-structured SEIR model with interventions.
+#'
+#' @export
+#'
+setMethod(
+  'transmission_parameters<-', 'SEIRDAge_interventions',
+  function(object, value) {
+    
+    # create list of parameter values
+    isolated_frac <- value$isolated_frac
+    beta_isolated <- value$beta_isolated
+    beta_not_isolated <- value$beta_not_isolated
+    kappa <- value$kappa
+    gamma <- value$gamma
+    mu <- value$mu
+    
+    trans_params <- list(isolated_frac, beta_isolated, beta_not_isolated, kappa, gamma, mu)
+    
+    # add names to each value
+    names(trans_params) = object@transmission_parameter_names
+    
+    # check format of parameters b, k and g
+    if(length(beta_isolated) != 1 | length(beta_not_isolated) != 1 | length(kappa) != 1 | length(gamma) != 1){
+      stop('The parameter values should be 1-dimensional.')
+    }
```

Comment:
If an age structured mu or isolated_frac is provided with the wrong number of age categories, the simulate fails rather ungracefully. 
The check in the normal age structured model seems sensible to me (make sure mu has length either 1 or the number of age categories)